src/controllers/product.controller.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Patch,
+  Post,
+  UseGuards,
+} from '@nestjs/common';
+import { AuthGuard } from '../auth/auth.guard';
+import { Product } from '../entities';
+import { ProductService } from '../services';
+import { ProductUpdateDto, ProductCreateDto } from '../dtos';
+
+@Controller('product')
```

Comment:
💭  something that we've been missing is the way the backend team defines the routes, in rest to define a route these should be in plural unless it's something unique in the project. I checked nestjs docs and this also applies in the framework. So something to keep in from now on is to create these routes as plural, later we can work on rewording those in the backend and frontend.

TL;DR this route should be `products` rather than `product`

```suggestion
@Controller('products')
```